libr/anal/fcn.c

```diff
@@ -76,12 +76,37 @@ R_API void r_anal_fcn_update_tinyrange_bbs(RAnalFunction *fcn) {
 	}
 }
 
+static void set_meta_min_if_needed(RAnalFunction *x) {
```

Comment:
Have you looked at what I've suggested above?

Reply:
Yes but the problem is that you can't key the tree on `addr` and then test to end the search by checking `rb_min_addr` [here](https://github.com/radare/radare2/pull/13176/files#diff-49a52329a1b3f5e5f62d571577391074R197), because there is no guarantee that `rb_min_addr` will be larger (or more accurately, `rb_min_addr` will not be smaller) as you climb the tree.

Reply:
> this is slow IMHO. In my PR i solved this by extending RTinyRange with a max_value, min_value fields.

Ok, lgtm

Reply:
> Yes but the problem is that you can't key the tree on addr and then test to end the search by checking rb_min_addr here, because there is no guarantee that rb_min_addr will be larger (or more accurately, rb_min_addr will not be smaller) as you climb the tree.

What do you mean? This has nothing to do with how you key the tree I think. What I'm saying is that you can already compute `min_value`/`max_value` (consequently, you can also compute _size) by computing those values as you construct the RTinyRange.

Reply:
Sorry, I was looking at all of the code at the given link, not just your suggestion.